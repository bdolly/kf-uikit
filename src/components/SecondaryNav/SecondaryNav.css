.SecondaryNav {
  @apply border-b-2
      relative
      bg-white
      overflow-hidden
      border-mediumGrey;
}

.SecondaryNav ul {
  @apply p-0
      m-0
      relative
      inline-block
      h-full
      max-w-full;
}

.SecondaryNav--link {
  @apply m-0
      p-0
      relative
      inline-block
      h-full
      border-lightGrey
      border
      border-b-0
      font-semibold
      cursor-pointer;

  a {
    @apply p-16
              block
              h-full
              text-sm
              font-semibold
              text-purple
              text-center
              font-title
              no-underline;
    &:focus {
      @apply text-pink
            border-b-4
            border-pink;
    }
  }
}

.SecondaryNav--link:hover a,
.SecondaryNav--link-active a {
  @apply text-pink
      border-b-4
      border-pink;
}
