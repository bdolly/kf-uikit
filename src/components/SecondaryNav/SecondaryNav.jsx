import React from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';
/**
 * Secondary navigation typically used under the header
 *
 * Must pass an array of NavLinks for the navigation bar.
 * Each SecondaryNavLink can take active, onNavClick as props, and {tabText}
 * as children.
 * `<SecondaryNavLink key={key} active={ifActive} onNavClick={() => {}} href={herf}>`
 * `{tabText}`
 * `</SecondaryNavLink>`
 */
const SecondaryNav = ({ buttons, className }) => {
  const navClass = classes('SecondaryNav', className);
  return (
    <nav className={navClass}>
      <ul className="SecondaryNav--list">{buttons ? buttons.map(button => button) : null}</ul>
    </nav>
  );
};

SecondaryNav.propTypes = {
  /** Any additional classes to be applied to secondary navbar */
  className: propTypes.string,
  /** NavLinks to display on the navbar */
  buttons: propTypes.node,
};

SecondaryNav.defaultProps = {
  className: null,
  buttons: null,
};

export default SecondaryNav;
