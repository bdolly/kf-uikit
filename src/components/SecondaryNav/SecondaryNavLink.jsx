import React from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';
/**
 * A navigation link item to be used inside the secondary nav bar
 *
 */
const SecondaryNavLink = ({ href, active, children, onNavClick, className }) => {
  const linkClass = classes(
    ['SecondaryNav--link', { 'SecondaryNav--link-active': active }],
    className,
  );
  return (
    <li className={linkClass}>
      <a href={href} onClick={onNavClick}>
        {children}
      </a>
    </li>
  );
};

SecondaryNavLink.propTypes = {
  /** Any additional classes to be applied to secondary navbar button */
  className: propTypes.string,
  /** Child elements to display on the navbar */
  children: propTypes.node,
  /** The url to navigate to */
  href: propTypes.string,
  /** A function to perform onClick */
  onNavClick: propTypes.func,
  /** Whether the link is active */
  active: propTypes.bool,
};

SecondaryNavLink.defaultProps = {
  className: null,
  children: null,
  active: false,
  onNavClick: () => {},
  href: null,
};

export default SecondaryNavLink;
