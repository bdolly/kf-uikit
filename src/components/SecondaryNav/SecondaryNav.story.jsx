/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, array } from '@storybook/addon-knobs';
import SecondaryNav from './SecondaryNav';
import SecondaryNavLink from './SecondaryNavLink';

const stories = storiesOf('SecondaryNav', module);

stories.addDecorator(withKnobs);

stories.add(
  'SecondaryNav',
  () => {
    const navList = array('Array of Tab Name', ['Tab1', 'Tab2', 'Tab3']);
    return (
      <div className="percy-min-width">
        <SecondaryNav
          buttons={navList.map(i => (
            <SecondaryNavLink
              key={i}
              active={navList.indexOf(i) === 0}
              onNavClick={e => {
                e.preventDefault();
              }}
              href="/"
            >
              {i}
            </SecondaryNavLink>
          ))}
        />
      </div>
    );
  },
  {
    info: {
      text: `
        The secondary nav is typically displayed below the primary navigation
        at the header of the site.
          `,
    },
  },
);
