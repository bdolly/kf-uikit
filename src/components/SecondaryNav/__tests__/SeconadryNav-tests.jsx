import React from 'react';
import renderer from 'react-test-renderer';
import SecondaryNav from '../SecondaryNav';
import SecondaryNavLink from '../SecondaryNavLink';

it(`secondary navigation renders correctly`, () => {
  const tree = renderer
    .create(
      <SecondaryNav
        buttons={[
          <SecondaryNavLink key={1} href="/">
            Testing
          </SecondaryNavLink>,
        ]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
