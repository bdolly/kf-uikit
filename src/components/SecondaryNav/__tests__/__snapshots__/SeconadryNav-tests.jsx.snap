// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`secondary navigation renders correctly 1`] = `
<nav
  className="SecondaryNav"
>
  <ul
    className="SecondaryNav--list"
  >
    <li
      className="SecondaryNav--link"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Testing
      </a>
    </li>
  </ul>
</nav>
`;
